src/main/java/de/eposcat/master/approachImpl/JSON_Oracle_DatabaseAdapter.java

```diff
@@ -82,6 +91,27 @@ public Page loadPage(int pageId) throws SQLException{
         }
     }
 
+    @Override
+    public List<Page> findPagesByType(String type) throws SQLException {
+        PreparedStatement stFindByType = conn.prepareStatement("SELECT * FROM pages WHERE type = ?");
```

Comment:
Missing null-check for the type parameter

Reply:
I'll rework both JSON adapters once the testing environments are ready.